netbox/netbox/views/__init__.py

```diff
@@ -167,27 +167,30 @@ def get(self, request):
             # If an object type has been specified, redirect to the dedicated view for it
             if form.cleaned_data['obj_type']:
                 object_type = form.cleaned_data['obj_type']
-                url = reverse(SEARCH_TYPES[object_type]['url'])
-                return redirect(f"{url}?q={form.cleaned_data['q']}")
-
-            for obj_type in SEARCH_TYPES.keys():
-
-                queryset = SEARCH_TYPES[obj_type]['queryset'].restrict(request.user, 'view')
-                filterset = SEARCH_TYPES[obj_type]['filterset']
-                table = SEARCH_TYPES[obj_type]['table']
-                url = SEARCH_TYPES[obj_type]['url']
-
-                # Construct the results table for this object type
-                filtered_queryset = filterset({'q': form.cleaned_data['q']}, queryset=queryset).qs
-                table = table(filtered_queryset, orderable=False)
-                table.paginate(per_page=SEARCH_MAX_RESULTS)
-
-                if table.page:
-                    results.append({
-                        'name': queryset.model._meta.verbose_name_plural,
-                        'table': table,
-                        'url': f"{reverse(url)}?q={form.cleaned_data.get('q')}"
-                    })
+                for _, value in SEARCH_TYPE_HIERARCHY.items():
```

Comment:
I understand that the for loop is not a very nice solution.

Regarding having the hierarchy separate:
I thought the fundamental problem of this bug is that 2 partially redundant data structures have to be maintained by hand.
Separating the hierarchy from the dict containing the search types still results in 2 different places holding the information. Or did I misunderstand you?

I could keep the `SEARCH_TYPE_HIERARCHY` as is, but add code to build a flattened version for building the view. This should solve the loop problem while not having to maintain partially redundant data structures by hand.

Reply:
I prepared it as described above, the `SEARCH_TYPE_HIERARCHY` still exists, and then `OBJECT_CHOICES` and `SEARCH_TYPES` are generated from that at program start.
The hierarchy is split into per-app dicts to keep in readable. That many layers of brackets and indentation would get to messy. Also the OrderedDicts are required because we might want to preserve the order in `OBJECT_CHOICES`

The result is that I don't even have to touch the code in the view and therefore not introduce a for-loop there.

Reply:
@jeremystretch This PR is waiting for a response from you.